### PR TITLE
WRO-745: Add spotlightId prop description for Panel and VirtualList 

### PR DESCRIPTION
**Pull request by jeonghee27**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
- There is no guide for specifying spotlightId when using VL in Panel. Only "id" is. 
- 'autofocus' guide is weird. 'PanelDecorator' looks like a prop name. because it has the wrong format.
- There is no guide for specifying "spotlightId" when using Panel without Panels.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Add `spotlightId` prop doc for VL / VGL. 
- Fix Panels format (add `autofocus'` name and add `PanelDecorator` docs )  
- Add `spotlightId` prop doc for Panel
- Fix `spotlightId` prop links

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-745

### Comments
Enact-DCO-1.0-Signed-off-by: Jeonghee Ahn (jeonghee27.ahn@lge.com)

### CODE REVIEW

File: Panels/Panel.js [hunk 1]
```diff
@@ -272,6 +283,17 @@ const PanelDecorator = hoc({defaultElement: `.${componentCss.body} *`}, (config,
...
  */
 
+/**
+ * The container id for {@link spotlight/Spotlight}.
+ *
+ * When used within [Panels]{@link sandstone/Panels.Panels}, this prop may be set by `Panels`.
+ *
+ * @name spotlightId
+ * @type {Boolean}
+ * @default false
```
Comment (seunghoh) [inline Panels/Panel.js]: String :)

File: Panels/Panel.js [hunk 2]
```diff
@@ -272,6 +283,17 @@ const PanelDecorator = hoc({defaultElement: `.${componentCss.body} *`}, (config,
  * @memberof sandstone/Panels.Panel.prototype
  */
 
+/**
+ * The container id for {@link spotlight/Spotlight}.
+ *
+ * When used within [Panels]{@link sandstone/Panels.Panels}, this prop may be set by `Panels`.
```
Comment (seunghoh) [inline Panels/Panel.js:289]: When the `Panel` is used within [Panels]{@link sandstone/Panels.Panels}, this prop will be generated by `Panels`.
When using the panel alone, you need to specify this prop for restoring focus.

File: VirtualList/VirtualList.js [hunk 3]
```diff
@@ -417,10 +417,15 @@ VirtualList.propTypes = /** @lends sandstone/VirtualList.VirtualList.prototype *
 	scrollMode: PropTypes.string,
 
 	/**
-	 * Spotlight Id.
+	 * The container id for [Spotlight container]@link spotlight/SpotlightContainerDecorator/#SpotlightContainerDecorator.spotlightId}.
+	 *
+	 * It is required to restore focus after `VirtualList` remount.
+	 *
+	 * For example, when the `VirtualList` is within a [Panel]{@link sandstone/Panels.Panel},,
```
Comment (seunghoh) [inline VirtualList/VirtualList.js:424]: ,, -> ,

File: VirtualList/VirtualList.js [hunk 4]
```diff
@@ -417,10 +417,15 @@ VirtualList.propTypes = /** @lends sandstone/VirtualList.VirtualList.prototype *
...
+	 * the `Spotlight` will store the id to restore focus when returning to the `Panel`.
```
Comment (seunghoh) [inline VirtualList/VirtualList.js:425]: the `Spotlight` will store the `spotlightId` to restore focus when returning to the `Panel`.

File: VirtualList/VirtualList.js [hunk 5]
```diff
@@ -889,11 +894,16 @@ VirtualGridList.propTypes = /** @lends sandstone/VirtualList.VirtualGridList.pro
...
-	 * Spotlight Id.
-	 *
-	 * @type {String}
-	 * @private
-	 */
+	* The container id for [Spotlight container]@link spotlight/SpotlightContainerDecorator/#SpotlightContainerDecorator.spotlightId}.
+	*
+	* It is required to restore focus after `VirtualList` remount.
+	*
+	* For example, when the `VirtualList` is within a [Panel]{@link sandstone/Panels.Panel},,
```
Comment (seunghoh) [inline VirtualList/VirtualList.js:901]: ,, -> ,

File: VirtualList/VirtualList.js [hunk 6]
```diff
@@ -889,11 +894,16 @@ VirtualGridList.propTypes = /** @lends sandstone/VirtualList.VirtualGridList.pro
...
+	* the `Spotlight` will store the id to restore focus when returning to the `Panel`.
```
Comment (seunghoh) [inline VirtualList/VirtualList.js:902]: the `Spotlight` will store the `spotlightId` to restore focus when returning to the `Panel`.

**Review by seunghoh — CHANGES_REQUESTED** [review]
Left some minor opinion.
I can't check the link in the doc. Please make sure the link works

File: Panels/Panel.js (hunk 1 above)
Comment (jeonghee27) [reply Panels/Panel.js]: I fixed

File: VirtualList/VirtualList.js (hunk 3 above)
Comment (jeonghee27) [reply VirtualList/VirtualList.js]: Sorry..

File: VirtualList/VirtualList.js (hunk 5 above)
Comment (jeonghee27) [reply VirtualList/VirtualList.js]: I fixed

File: VirtualList/VirtualList.js (hunk 4 above)
Comment (jeonghee27) [reply VirtualList/VirtualList.js]: I fixed

File: VirtualList/VirtualList.js (hunk 6 above)
Comment (jeonghee27) [reply VirtualList/VirtualList.js]: I fixed

File: Panels/Panel.js (hunk 2 above)
Comment (jeonghee27) [reply Panels/Panel.js]: Thanks I fixed

**Review by seunghoh — APPROVED** [review]

**Comment by seunghoh** [comment]
LGTM

File: VirtualList/VirtualList.js [hunk 7]
```diff
@@ -889,11 +894,16 @@ VirtualGridList.propTypes = /** @lends sandstone/VirtualList.VirtualGridList.pro
...
 
 	/**
-	 * Spotlight Id.
-	 *
-	 * @type {String}
-	 * @private
-	 */
+	* The container id for [Spotlight container]{@link spotlight/SpotlightContainerDecorator/#SpotlightContainerDecorator.spotlightId}.
+	*
+	* It is required to restore focus after `VirtualList` remount.
```
Comment (MikyungKim) [inline VirtualList/VirtualList.js:899]: VirtualGridList

Comment (jeonghee27) [reply VirtualList/VirtualList.js]: I fixed.

File: VirtualList/VirtualList.js [hunk 8]
```diff
@@ -417,10 +417,15 @@ VirtualList.propTypes = /** @lends sandstone/VirtualList.VirtualList.prototype *
 	scrollMode: PropTypes.string,
 
 	/**
-	 * Spotlight Id.
+	 * The container id for [Spotlight container]{@link spotlight/SpotlightContainerDecorator/#SpotlightContainerDecorator.spotlightId}.
+	 *
+	 * It is required to restore focus after `VirtualList` remount.
```
[the remark this answers is not in the document]
Comment (jeonghee27) [reply VirtualList/VirtualList.js]: I fixed.

File: VirtualList/VirtualList.js [hunk 9]
```diff
@@ -417,10 +417,15 @@ VirtualList.propTypes = /** @lends sandstone/VirtualList.VirtualList.prototype *
...
+	 *
+	 * For example, when the `VirtualList` is within a [Panel]{@link sandstone/Panels.Panel},
+	 * the `Spotlight` will store the `spotlightId` to restore focus when returning to the `Panel`.
```
[the remark this answers is not in the document]
Comment (jeonghee27) [reply VirtualList/VirtualList.js]: I fixed. Thanks

**Review by MikyungKim — APPROVED** [review]
Great! :lgtm:

**Pull request merged**